samza-core/src/main/java/org/apache/samza/runtime/LocalApplicationRunner.java

```diff
@@ -272,4 +285,44 @@ StreamProcessor createStreamProcessor(
           taskFactory.getClass().getCanonicalName()));
     }
   }
+
+  /**
+   * Generates a unique latch id that is consistent across processors within the same application lifecycle.
+   * Each {@code StreamSpec} has an id that is unique and is used for hashcode computation.
+   *
+   * @param intStreams list of {@link StreamSpec}s
+   * @return latch id
+   */
+  private String generateLatchId(List<StreamSpec> intStreams) {
+    return String.valueOf(Objects.hashCode(intStreams.stream()
```

Comment:
Do we know that the hashcode is the same for lists with the same elements but in different orders? Is there any assumption in this code that the intStreams is a sorted list? Would be nice to see a test case for this code.